### DISCUSSION TITLE
Client libraries for ros-foxglove-bridge

### DISCUSSION THREAD

**Discussion by rosterloh** [foxglove/community#631] [Q&A]
While I love your product foxglove-studio I was wondering if it was possible to use the ros-foxglove-bridge in my own application. I've used rosbridge with roslibpy and could benefit from the perfomace imporvements of ros-foxglove-bridge. Would I need to write my own client library in this case from the protocol spec?

**Comment by jtbandes** [comment]
We have a TypeScript client available here: https://github.com/foxglove/ws-protocol
There is currently no reference implementation for a Python client, but it would be a reasonable goal to add one here.

Please note that some protocol enhancements have been added recently and not all the reference implementations may have the latest features.

**Reply by defunctzombie** [reply]
For ros folks we could provide a package built on top of the ws-protocol client libraries which handles the de/serialization boilerplate. Or at a minimum an example (if we don't have one already).

**Comment by EnzoGhisoni** [comment]
Is there some news about a python client example or library ?
